src/addons/shallow-diff.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Diffing algorithm for presenters. We need to recalculate the
+ * model of props are state change, but ignore children.
+ */
+
+export default function shallowDiff(a, b) {
```

Comment:
should this be called `isShallowDiff` or `shallowDifferent`. Or maybe flip the logic and call it `shallowEqual` ?
Having it end in just `diff` makes me think it's going to return a set of items that are different (like lodash `difference`) or a file change diff

Reply:
Ah. Good call. This is like "Shallow equal, but not children". Maybe this should be `shouldUpdateModel`

Reply:
also, if it's used internally only, should it live in `src/addons/`?